sdk/scheduler/build.gradle

```diff
@@ -1,149 +1,173 @@
-// Config specific to the dcos-commons library:
-apply plugin: 'application'
+plugins {
+    id 'de.fuerstenau.buildconfig' version '1.1.8'
+    id 'com.github.johnrengelman.shadow' version '2.0.3'
+}
 
-ext {
-    antlrVer = "4.5.1-1"
-    commonsCollectionsVer = "3.2.2"
-    commonsIoVer = "2.4"
-    curatorVer = "2.9.1"
-    dropwizardVer = "0.9.2"
-    findbugsAnnotationsVer = "3.0.1u2"
-    hamcrestVer = "1.3"
-    jacksonVer = "2.6.3"
-    jacksonProtobufVer = "0.9.3"
-    junitVer = "4.12"
-    slf4jVer = "1.7.10"
-    jsonVer = "20160212"
-    protobufFormatVer = "1.4"
-    mockitoVer = "1.9.5"
-    restServiceVer = "2.0.1"
-    slf4jVer = "1.7.10"
-    systemRulesVer = "1.16.0"
-    mockServerVer = "3.10.4"
-    springVer = "4.3.1.RELEASE"
-    httpClientVer = "4.5.2"
-    awaitilityVer = "2.0.0"
-    javaDiffUtilsVer = "1.3.0"
-    jettyServletVer = "9.4.9.v20180320"
-    jerseyVer = "2.23"
-    mimepullVer = "1.9.6"
-    jsonProviderVer = "2.2.3"
-    mustacheVer = "0.9.2"
-    hibernateValidatorVer = "5.3.2.Final"
-    elVer = "2.2.4"
-    jwtVer = "3.2.0"
-    bouncyCastleVer = "1.57"
-    j2htmlVer = "1.2.0"
-    dropWizardMetricsVer = "3.2.5"
+// Configure the SDKBuildInfo class generator
+def getGitHash = { ->
+    def stdout = new ByteArrayOutputStream()
+    exec {
+        commandLine 'git', 'rev-parse', '--short', 'HEAD'
+        standardOutput = stdout
+    }
+    exec {
+        // Teamcity workaround: using 'git describe' to determine dirty status doesn't work there.
+        commandLine '/bin/sh', '-c', 'if [ -n "$(git diff --name-only HEAD)" ]; then echo -dirty; fi'
+        standardOutput = stdout
+    }
+    return stdout.toString().replace("\n", "")
+}
+buildConfig {
+    packageName = 'com.mesosphere.sdk.generated'
+    clsName = 'SDKBuildInfo'
+
+    version = project.version // VERSION
+    appName = project.group + '.' + project.name // NAME
+    buildConfigField 'String', 'GIT_SHA', "${getGitHash()}"
+    buildConfigField 'long', 'BUILD_TIME_EPOCH_MS', System.currentTimeMillis() + 'L'
 }
 
 task sourceJar(type: Jar) {
     from sourceSets.main.allJava
+    classifier 'sources'
 }
+shadowJar {
+    classifier = ''
+
+    mergeServiceFiles()
+
+    // Rename the following packages so that they do not conflict with downstream builds.
+    // For now, this list specifically consists of libraries that collide within the Jenkins Mesos
+    // plugin, but other cases may be added as needed.
+
+    // io.dropwizard.metrics:metrics-core
```

Comment:
Want to remove all the commented packages?

Reply:
I left these here to list the packages that were being affected by the rename